wiring/inc/spark_wiring_wifitester.h

```diff
@@ -0,0 +1,112 @@
+  version 3 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************
+ */
+
+#ifndef WIFITESTER_H
+#define WIFITESTER_H
+
+#include "system_setup.h"
+#include "spark_wiring_startup.h"
+
+#ifdef __cplusplus
+
+class WiFiTester
+{
+    bool useSerial1;
+
+    uint8_t notifiedAboutDHCP;
+    int state;
+    int wifi_testing;
+    int dhcp_notices;
+    unsigned cmd_index;
+    static const unsigned cmd_length = 256;
+    char command[cmd_length];
+    bool power_state = false;
+
+
+    void checkWifiSerial(char c);
+    void wifiScan();
+    void printInfo();
+    void printItem(const char* name, const char* value);
+    void tester_connect(char *ssid, char *pass);
+    void tokenizeCommand(char *cmd, char* parts[], unsigned max_parts);
+    bool isPowerOn();
+
+public:
+    WiFiTester() {
+        memset(this, 0, sizeof(*this));
+    }
+
+    void setup(bool useSerial1);
+    void loop(int c);
+
+    uint8_t serialAvailable();
+    int32_t serialRead();
+    void serialPrintln(const char* s);
+    void serialPrint(const char* s);
+};
+
+#if defined(SETUP_OVER_SERIAL1) && SETUP_OVER_SERIAL1 == 1
+
+#define SETUP_WIFITESTER_IMPL() \
+    { \
+        system_tester_handlers_t handlers = {0}; \
+        handlers.version = 1; \
+        handlers.size = sizeof(handlers); \
+        handlers.create = [](void* reserved) -> void* { \
+            return new WiFiTester(); \
+        }; \
+        handlers.destroy = [](void* tester, void* reserved) -> int { \
+            WiFiTester* t = reinterpret_cast<WiFiTester*>(tester); \
+            if (t != nullptr) { \
+                delete t; \
+                return 0; \
+            } \
+            return 1; \
+        }; \
+        handlers.setup = [](void* tester, bool useSerial1, void* reserved) -> int { \
+            WiFiTester* t = reinterpret_cast<WiFiTester*>(tester); \
+            if (t != nullptr) { \
+                t->setup(useSerial1); \
+                return 0; \
+            } \
+            return 1; \
+        }; \
+        handlers.loop = [](void* tester, int c, void* reserved) -> int { \
+            WiFiTester* t = reinterpret_cast<WiFiTester*>(tester); \
+            if (t != nullptr) { \
+                t->loop(c); \
+                return 0; \
+            } \
+            return 1; \
+        }; \
+        system_set_tester_handlers(&handlers, nullptr); \
+    }
+#else
+#define SETUP_WIFITESTER_IMPL()
+#endif
+
+#define SETUP_WIFITESTER() STARTUP(SETUP_WIFITESTER_IMPL())
```

Comment:
ping @avtolstoy - any thoughts on that?

Reply:
`SETUP_WIFITESTER()` is not used and instead we now enable it via `STARTUP(System.enableFeature(FEATURE_WIFITESTER));`, so it can be removed.

I'll try moving the implementation into a function today.